hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java

```diff
@@ -620,8 +620,7 @@ public synchronized boolean expireServer(final ServerName serverName) {
     }
   }
 
-  @VisibleForTesting
-  public void moveFromOnlineToDeadServers(final ServerName sn) {
+  public synchronized void moveFromOnlineToDeadServers(final ServerName sn) {
     synchronized (onlineServers) {
```

Comment:
The locking here is a bit strange, onlineServers is already a ConcurrentMap, and seems this is the only place where we synchronized on it. Not your fault, can review it later.